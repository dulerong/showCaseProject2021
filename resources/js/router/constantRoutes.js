import Vue from 'vue'

export const routePath = {
  home: {
    title: 'Home',
    path: '',
    name: 'home',
    icon: 'mdi-home'
  },
  search: {
    title: 'Search',
    path: '/search',
    name: 'search',
    icon: 'mdi-magnify'
  },
  manage: {
    title: 'Manage',
    path: '/manage',
    name: 'manage',
    icon: 'mdi-alpha-m-circle'
  },
  stat: {
    title: 'Statistics',
    path: 'stat',
    name: 'stat',
    icon: 'mdi-file-chart'
  },
  login: {
    title: 'LOGIN',
    path: 'login', 
    name: 'login',
    icon: 'mdi-login'
  },
  forgotPassword: {
    title: null,
    path: '/forgot-password',
    name: 'forgotPassword',
    icon: null
  },
  memberPasswordReset: {
    title: 'Password',
    path: '/manage-password',
    name: 'managePassword',
    icon: 'mdi-shield-key'
  },
  test: {
    title: 'test',
    path: '/test',
    name: 'test',
    icon: 'calendar'
  }
}
Vue.prototype.$_routePath = routePath

export const navigation = [
  {
    title: 'GENERAL',
    children: [
      routePath.home,
      routePath.stat,
    ]
  },
  // {
  //   title: '集計管理',
  //   children: [
  //     routePath.statpersonal,
  //     routePath.statbranch,
  //     routePath.statbranchemployee,
  //     routePath.statInsideAll,
  //     routePath.statInsideBranch,
  //   ]
  // },
  // {
  //   title: 'イベント管理',
  //   children: [
  //     routePath.calendar,
  //   ]
  // },
  // {
  //   title: 'OTHERS',
  //   children: [
  //     routePath.memberPasswordReset,
  //   ]
  // },
  // {
  //   title: '暫定的（削除予定）',
  //   children: [
  //     routePath.test
  //   ]
  // }
]
Vue.prototype.$_navigation = navigation

export const footer_navigation = [
  routePath.home,
  routePath.stat,
]
Vue.prototype.$_footer_navigation = footer_navigation