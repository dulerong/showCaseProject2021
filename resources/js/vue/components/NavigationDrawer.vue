<template lang="pug">
v-navigation-drawer(
  app
  height="100%"
  :absolute="$vuetify.breakpoint.mdAndUp"
  :permanent="$vuetify.breakpoint.mdAndUp"
  :mini-variant="drawer.isMini && $vuetify.breakpoint.mdAndUp"
  v-model="drawer.isVisible"
)
  v-list-item
    v-list-item-icon(class="mr-2")
      v-icon mdi-desktop-classic
    v-list-item-content
      v-list-item-title(class="text-h6") {{ $t("naviDrawer.mainTitle") }}
      v-list-item-subtitle {{ $t("naviDrawer.subTitle") }}
  v-divider
  v-list
    v-list-item(
      v-for="(tab, i) in $_navigation[0].children"
      :key="i"
      :to="{ name: tab.name, params: { lang: $route.params.lang } }"
      exact
    )
      v-list-item-icon(class="mr-2")
        v-icon {{ tab.icon }}
      v-list-item-content
        v-list-item-title(
          v-text="$t(`naviDrawer.GENERAL.children.${tab.name}`)"
        )
    //- Below are code I wrote for subgroup, that I no longer use now
    //- v-list-group(v-for="(subgroup, i) in $_navigation" :key="i" :value="false")
    //-   template(v-slot:activator="")
    //-     v-list-item-title(v-text="$t(`naviDrawer.${subgroup.title}.name`)")
    //-   v-list-item(v-for="(tab, i) in subgroup.children" :key="i" :to="{ name: tab.name, params: { lang: $route.params.lang } }" exact)
    //-     v-list-item-icon(class="mr-2")
    //-       v-icon(style="width: 20px") {{ tab.icon }}
    //-     v-list-item-content
    //-       v-list-item-title(v-text="$t(`naviDrawer.${subgroup.title}.children.${tab.name}`)")
</template>

<script>
export default {
  props: {
    drawer: {
      type: Object,
      required: true,
      isMini: { type: Boolean, required: true },
      isVisible: { type: Boolean, required: true }
    }
  },
};
</script>