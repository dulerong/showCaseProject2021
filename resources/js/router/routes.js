// route paths
import { routePath } from './constantRoutes'

// layouts
import DefaultLayout from '~layouts/default'
import LoginLayout from '~layouts/login'

// pages
import HomePage from '~pages/HomePage'
import LoginPage from '~pages/LoginPage'
import StatPage from '~pages/StatPage'
import NotFoundPage from '~pages/NotFoundPage'

// import { store } from './store'

export default [
  {
    path: routePath.home.path,
    name: routePath.home.name,
    component: HomePage,
    meta: { layout: DefaultLayout }
  },
  {
    path: routePath.stat.path,
    name: routePath.stat.name,
    component: StatPage,
    meta: { layout: DefaultLayout }
  },
  {
    path: routePath.login.path,
    name: routePath.login.name,
    component: LoginPage,
    meta: { layout: LoginLayout }
  },
  {
    path: '*',
    component: NotFoundPage,
    meta: { layout: DefaultLayout, requiresAuth: true }
  }
]