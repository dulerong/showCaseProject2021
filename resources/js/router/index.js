import Vue from 'vue'
import VueRouter from 'vue-router'
import routes from './routes'
import i18n from '../i18n'
import vuetify from '../vuetify'
// import { store } from './store'
// import { routePath } from './constantsRoutes'

Vue.use(VueRouter)

const router = new VueRouter({
  routes,
  mode: 'history',
  scrollBehavior() {
    return { x: 0, y: 0, behavior: 'smooth' }
  },
})

const langList = ['en', 'ch']

router.beforeEach((to, from, next) => {
  const { lang } = to.params
  if (!langList.includes(lang)) return next('en') // if requested language not available, set locale to english and redirect to homepage

  i18n.locale = lang // set current language for i18n.
  vuetify.framework.lang.current = lang // set current language for vuetify
  next()

  // if (to.path === '/') next({ name: 'login', params: { lang: 'en' } })
  // if (!langList.includes(lang)) next({ name: to.name, params: { lang: 'en' } })
  // else next()
})

// router.beforeEach((to, from, next) => {
//   if (to.meta.layout !== from.meta.layout) store.commit('notification/hideNotification')

//   if (to.matched.some(record => record.meta.requiresAuth)) {
//     store.getters['auth/isLoggedIn'] ? next() : next({ name: routePath.login.name })
//   } else if (to.matched.some(record => record.meta.requiresVisitor)) {
//     store.getters['auth/isLoggedIn'] ? next({ name: routePath.home.name }) : next()
//   } else {
//     next()
//   }
// })
  
export default router