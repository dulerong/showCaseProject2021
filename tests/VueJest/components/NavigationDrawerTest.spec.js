import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'

import NavigationDrawer from '~components/NavigationDrawer'
import { navigation } from '~js/router/constantRoutes'

describe('AppBar Component', () => {
  const localVue = createLocalVue()

  let vuetify
  
  beforeEach(() => {
    vuetify = new Vuetify()
  })

  afterEach(() => {    
    jest.clearAllMocks()
  })

  const mountFunction = (isMini = false, options) => {
    return mount(NavigationDrawer, {
      localVue,
      vuetify,
      mocks: {
        $t: (msg) => msg,
        $route: {
          params: { lang: 'en' }
        }
      },
      stubs: ['router-link'],
      propsData: {
        drawer: {
          isMini
        }
      },
      ...options
    })
  }

  it('Renders properly', () => {
    const wrapper = mountFunction(true)

    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Opens when passed isMini = true', () => {
    const wrapper = mountFunction(true)

    const naviDrawer = wrapper.find('.v-navigation-drawer--close')
    expect(naviDrawer.exists()).toBeFalsy()  
  })

  it('Contains correct amount of groups', () => {
    const wrapper = mountFunction()

    const group = wrapper.findAll('router-link-stub')
    expect(group.length).toEqual(navigation[0].children.length)
  })
})