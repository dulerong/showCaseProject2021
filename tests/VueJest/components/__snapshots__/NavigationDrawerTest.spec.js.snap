// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar Component Renders properly 1`] = `
<nav class="v-navigation-drawer v-navigation-drawer--absolute v-navigation-drawer--mini-variant v-navigation-drawer--open theme--light" style="height: 100%; top: 0px; max-height: calc(100% - 0px); transform: translateX(0%); width: 56px;">
  <div class="v-navigation-drawer__content">
    <div tabindex="-1" class="v-list-item theme--light">
      <div class="v-list-item__icon mr-2"><i aria-hidden="true" class="v-icon notranslate mdi mdi-desktop-classic theme--light"></i></div>
      <div class="v-list-item__content">
        <div class="v-list-item__title text-h6">naviDrawer.mainTitle</div>
        <div class="v-list-item__subtitle">naviDrawer.subTitle</div>
      </div>
    </div>
    <hr role="separator" aria-orientation="horizontal" class="v-divider theme--light">
    <div class="v-list v-sheet theme--light">
      <router-link-stub tabindex="0" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__icon mr-2"><i aria-hidden="true" class="v-icon notranslate mdi mdi-home theme--light"></i></div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">naviDrawer.GENERAL.children.home</div>
        </div>
      </router-link-stub>
      <router-link-stub tabindex="0" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__icon mr-2"><i aria-hidden="true" class="v-icon notranslate mdi mdi-file-chart theme--light"></i></div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">naviDrawer.GENERAL.children.stat</div>
        </div>
      </router-link-stub>
      <router-link-stub tabindex="0" class="v-list-item v-list-item--link theme--light">
        <div class="v-list-item__icon mr-2"><i aria-hidden="true" class="v-icon notranslate mdi mdi-baby-face-outline theme--light"></i></div>
        <div class="v-list-item__content">
          <div class="v-list-item__title">naviDrawer.GENERAL.children.japanPopulation</div>
        </div>
      </router-link-stub>
    </div>
  </div>
  <div class="v-navigation-drawer__border"></div>
</nav>
`;
